kernel/lib/putchar.cpp

```diff
@@ -41,11 +41,21 @@ uint16_t px_ansi_vga_table[16] = {
     VGA_LightCyan, VGA_White
 };
 // Printing mutual exclusion
-px_mutex_t put_mutex;
+px_mutex_t put_mutex("putc");
 
-int putchar(char c) {
+int putchar(char c)
+{
+    int retval;
     // Gain mutual exclusion
     px_mutex_lock(&put_mutex);
+    retval = putchar_unlocked(c);
+    // Release mutual exclusion
+    px_mutex_unlock(&put_mutex);
+    return retval;
+}
+
+int putchar_unlocked(char c) {
```

Comment:
I get why you named this function `putchar_unlocked`, but I feel like that's a bad name given that there are locks around it in other functions (like `putchar`).

Reply:
Yeah I do see how that could be confusing, but I do feel like this is good name. I'm totally open to changing the name, but I can't think of something at the moment.

Reply:
Would `putchar_internal` work? Or something like that?

Reply:
Maybe, but it's not actually internal. I feel like unlocked is really the best name for it. Maybe I can just add some documentation to that function explaining the name?

Reply:
Fair enough. I'll resolve it.